components/Navbar/NavbarComponentsBuilder.tsx

```diff
@@ -0,0 +1,66 @@
+import React from "react";
+import prisma, { Prisma } from "@/lib/prisma";
+import { NavbarItemBuilder } from "./NavbarItem";
+import { NavbarDropdownBuilder } from "./NavbarDropdownBuilder";
+import { ImageComponent } from "../Shared/ImageComponent";
+import { getAssetPath } from "@/lib/getAssetInfo";
+import { NavbarComponents } from "./NavbarComponentsComponent";
+
+type NavbarComponentsBuilderProps = Prisma.NavbarGetPayload<{}>;
+type NavbarComponentBuilderProps = Prisma.NavbarComponentsGetPayload<{}>;
+
+export async function NavbarComponentsBuilder({
+  dateUpdated,
+  logo,
+  id,
+  userUpdated,
+}: NavbarComponentsBuilderProps) {
+  const navbarComponents = await prisma.navbarComponents.findMany({
+    orderBy: {
+      sort: "asc",
+    },
+  });
+  const navbarBuiltComponents = navbarComponents.map(NavbarComponentBuilder);
+  let navbarInfo: NavbarComponentsBuilderProps = {
+    dateUpdated,
+    id,
+    logo,
+    userUpdated,
+  };
+  if (logo)
+    navbarInfo = {
+      dateUpdated,
+      logo: await getAssetPath(logo),
+      id,
+      userUpdated,
+    };
+
+  return (
+    <NavbarComponents componentsSlot={navbarBuiltComponents} {...navbarInfo} />
+  );
+}
+
+const NavbarComponentBuilder = (
+  navbarComponent: NavbarComponentBuilderProps
+) => {
+  if (navbarComponent.item === null) return null;
+
+  let itemIdInt = Number(navbarComponent.item);
+
+  switch (navbarComponent.collection) {
+    case "navbar_items":
+      return (
+        <NavbarItemBuilder id={itemIdInt} key={String(navbarComponent.id)} />
+      );
+
+    case "navbar_dropdowns":
+      return <NavbarDropdownBuilder id={itemIdInt} key={navbarComponent.id} />;
+
+    default:
+      return (
+        <div title={JSON.stringify(navbarComponent, null, 2)}>
+          Unknown collection
+        </div>
+      );
+  }
```

Comment:
Let's use the new way to do this without the switch/case. [See here](https://github.com/PfadiMH/site/commit/47a0d5f9a42347d79770a9d6d07986f8eb40d45a#diff-c179c7f38284fb21257afa6bfe339b18c7ad989ecbe01787a1337112c828c07fR64-R81)

Reply:
Post MVP